### PR TITLE
cdb2sql: send cancel cnonce to connected db when hitting ctrl-c

### PR DESCRIPTION
**Pull request by adizaimi**

### CODE REVIEW

File: sqlite/func.c [hunk 1]
```diff
@@ -416,7 +416,7 @@ static void sleepFunc(sqlite3_context *context, int argc, sqlite3_value *argv[])
   for(i = 0; i < n; i++) {
     sleep(1);
     if( comdb2_sql_tick() )
-      break;
+      break;  /* COMDB2: we could also return error sqlite3_result_error(context, "Interrupted", -1); */
```
Comment (nirbhayc) [inline sqlite/func.c]: You will get a formatting error here. Please break the comment.

File: tools/cdb2sql/cdb2sql.c [hunk 2]
```diff
@@ -694,7 +695,9 @@ static void process_line(char *sql, int ntypes, int *types)
     sqlstr[len] = '\0';
 
     int start_time_ms, run_time_ms;
+    gbl_in_stmt = 1;
```
Comment (nirbhayc) [inline tools/cdb2sql/cdb2sql.c]: Use of gbl_in_stmt seems redundant to me. If we have a nonce, let's send it to the server and let the server take care of terminating the query if a match found.

File: tools/cdb2sql/cdb2sql.c [hunk 3]
```diff
@@ -887,6 +890,38 @@ static void replace_args(int argc, char *argv[])
...
+    cdb2_hndl_tp *cdb2h_2 = NULL;
+    int rc;
+    if (dbhostname) {
+        rc = cdb2_open(&cdb2h_2, dbname, dbhostname, CDB2_DIRECT_CPU);
+    } else {
+        rc = cdb2_open(&cdb2h_2, dbname, dbtype, 0);
+    }
+    if(rc) { 
+        if (debug_trace)
+            printf("Open db to cancel sql failed\n");
```
Comment (nirbhayc) [inline tools/cdb2sql/cdb2sql.c]: It should be printed to stderr and please also print the return code.

File: sqlite/func.c (hunk 1 above)
Comment (adizaimi) [inline sqlite/func.c]: sqlite has its own formatting larger than 80 chars. This should be fine.

File: tools/cdb2sql/cdb2sql.c (hunk 2 above)
Comment (adizaimi) [inline tools/cdb2sql/cdb2sql.c]: It is useless to talk to the server if there is no query actually running. Imagine doing ctrl_c several times in a row--you would talk to the server trying to cancel something without good reason.

Comment (nirbhayc) [inline tools/cdb2sql/cdb2sql.c]:
```
Imagine doing ctrl_c several times in a row--you would talk to the server trying to cancel something without good reason.
```
That's not right. The first Ctrl-C should kill the query and the next one should kill the client.

Comment (adizaimi) [inline tools/cdb2sql/cdb2sql.c]: What you are explaining is what I had implemented, until the consensus was to behave like bash and allow ctrl_c in a row to do nothing but just go to newline. Thus i removed that behavior.

Comment (nirbhayc) [inline tools/cdb2sql/cdb2sql.c]: oh ok. got a little confused.

**Pull request merged**